packages/server/src/automations/tests/scenarios/looping.spec.ts

```diff
@@ -152,6 +152,44 @@ describe("Loop automations", () => {
     )
   })
 
+  it("ensure the loop stops if the max iterations are reached", async () => {
+    const builder = createAutomationBuilder({
+      name: "Test Loop max iterations",
+    })
+
+    const results = await builder
+      .appAction({ fields: {} })
+      .loop({
+        option: LoopStepType.ARRAY,
+        binding: ["test", "test2", "test3"],
+        iterations: 2,
+      })
+      .serverLog({ text: "{{loop.currentItem}}" })
+      .serverLog({ text: "{{steps.1.iterations}}" })
+      .run()
+
+    expect(results.steps[1].outputs.iterations).toBe(2)
```

Comment:
So, just a little weirdness with our automation system here that you've fallen victim to. This should actually be

`expect(results.steps[0].outputs.iterations).toBe(2)`

As we don't count a loop as a _real_ step. But within the handlebars that's correct as the trigger is counted. I will document this somewhere in the test builder to avoid this confusion, my fault!